shared/actions/wallets.js

```diff
@@ -10,6 +10,42 @@ import logger from '../logger'
 import type {TypedState} from '../constants/reducer'
 import {walletsTab} from '../constants/tabs'
 
+const buildPayment = (state: TypedState) =>
+  RPCTypes.localBuildPaymentLocalRpcPromise({
+    amount: state.wallets.get('buildingPayment').get('amount'),
+    // FIXME: Assumes XLM.
+    from: state.wallets.selectedAccount,
+    fromSeqno: '',
+    publicMemo: state.wallets.get('buildingPayment').get('publicMemo'),
+    secretNote: state.wallets.get('buildingPayment').get('secretNote').stringValue(),
+    to: state.wallets.get('buildingPayment').get('to'),
+    toIsAccountID: true,
+  }).then(build => WalletsGen.createBuiltPaymentReceived({
+      build: Constants.buildPaymentResultToBuiltPayment(build),
+    })
+  )
+
+const sendPayment = (state: TypedState) =>
+  RPCTypes.localSendPaymentLocalRpcPromise({
+    amount: state.wallets.get('buildingPayment').get('amount'),
+    // FIXME -- support other assets.
+    asset: {type: 'native', code: '', issuer: ''},
+    from: state.wallets.get('buildingPayment').get('from'),
+    fromSeqno: '',
+    publicMemo: state.wallets.get('buildingPayment').get('publicMemo'),
+    quickReturn: false,
+    secretNote: state.wallets.get('buildingPayment').get('secretNote').stringValue(),
+    to: state.wallets.get('buildingPayment').get('to'),
+    toIsAccountID: !!state.wallets.get('builtPayment').get('toUsername'),
+    worthAmount: '',
+  }, Constants.sendPaymentWaitingKey).then(res =>
+    WalletsGen.createClearBuildingPayment()
```

Comment:
We shouldn't chain these actions like this. This should return the one action and those should be side effects